### ISSUE TITLE
Select box immediatly closed when space pressed

### ISSUE THREAD

**Issue by isidorn** [microsoft/vscode#103496]
Windows or Linux - platforms where we use the custom select box

1. Output channel, shift + tab to focus on the output channel dropdown
2. Press `space` -> the select box opens and immeadtly closes

This was working in older stables, but I can reproduce with latest stable and insiders. it is a regression

**Comment by isidorn** [comment]
I have investigated into this and the issue only present in the Output view.
The problem is that on pressing `space` we invoke the `switchBetweenOutputs` action from [here](https://github.com/microsoft/vscode/blob/15e798e081984a17eae00de91adf4479190d5ed2/src/vs/workbench/contrib/output/browser/output.contribution.ts#L118)

The fix which I found was to simply pass preserveFocus `true` [here](https://github.com/microsoft/vscode/blob/15e798e081984a17eae00de91adf4479190d5ed2/src/vs/workbench/contrib/output/browser/output.contribution.ts#L118)

However the proper fix would be if that action is not triggered on space. Thus also assigning to @sandy081 to get his thoughts

fyi @webczat

**Comment by sandy081** [comment]
Thanks for investigating @isidorn. 

Agreed that the right fix would be not to run the action on space. But may I know who/why this action is running on space? I thought this should not happen. There is an action item implementing this action here - 

https://github.com/microsoft/vscode/blob/1858765/src/vs/workbench/contrib/output/browser/outputView.ts#L270

Probably `space` event has to be stopped propagating ? I can't reproduce this on Mac and other platforms use custom select box

https://github.com/microsoft/vscode/blob/1858765/src/vs/base/browser/ui/selectBox/selectBoxCustom.ts#L218

**Comment by isidorn** [comment]
@sandy081 to repro on mac, just change this code here to make sure to use the custom select box 
https://github.com/microsoft/vscode/blob/15e410b98b88ca11bd3b059c5b7eec04b08693ce/src/vs/base/browser/ui/selectBox/selectBox.ts#L84

Yeah, I believe deep down there might be an issue with the action item. Since when an action item is present, the space should not trigger the action. I have no clue why this only happens in the outline view.

**Comment by isidorn** [comment]
CustomSelectBox was reacting on KEY_DOWN and preventing it, while actionbar is reacting on KEY_UP, thus both would get triggered.
Changed CustomSelectBox to react on KEY_UP

**Comment by isidorn** [comment]
Unfortunetly will have to reopen this one. The fix I did caused regressions in other areas.

Thus my proposed fix is to move away from custom dropdowns and use native dropdowns on all platforms, which we will probably look into in August.
@webczat as a workaround. F1 > type `view `  and you can choose which output view to naviagate to

**Comment by isidorn** [comment]
@sandy081 I actually found a neat fix for this. When the action is triggered in an invalid way the channelId is not passed. So just checking the channelId if it is a string seems to nicely fix the issue and does not break anything.
Let me know what you think. I was unable to fix this with changes in the custom dropdown, and I think this is not a bad fix.

**Comment by sandy081** [comment]
@isidorn Thanks for the fix.
